### PR TITLE
KOGITO-5553: Data-index API connection with runtimes: Add Node Instances operations and process variables update

### PR DESCRIPTION
**Pull request by nmirasch**
Depends on https://github.com/kiegroup/kogito-runtimes/pull/1563

jira: https://issues.redhat.com/browse/KOGITO-5553

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Comment by nmirasch** [comment]
jenkins please retest this

File: data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java [hunk 1]
```diff
@@ -136,6 +171,24 @@ protected CompletableFuture sendPostClientRequest(WebClient webClient, String re
...
+                    .putHeader("Authorization", getAuthHeader())
+                    .sendJson(new JsonObject(jsonBody), res -> {
+                        if (res.succeeded() && (res.result().statusCode() == 200)) {
+                            future.complete(res.result().bodyAsString());
+                        } else {
+                            future.completeExceptionally(new DataIndexServiceException(getErrorMessage(logMessage, res.result())));
+                        }
+                    });
+        } catch (Exception e) {
+            new DataIndexServiceException("FAILED: " + logMessage + " errorMEssage:" + e.getMessage());
```
Comment (cristianonicolai) [inline data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java]: maybe
```suggestion
            new DataIndexServiceException("Failed to " + logMessage + ", e);
```

File: data-index/data-index-service/data-index-service-common/src/main/resources/schema.graphqls [hunk 2]
```diff
@@ -16,6 +16,10 @@ type Mutation {
     ProcessInstanceAbort(id: String): String
     ProcessInstanceRetry(id: String): String
     ProcessInstanceSkip(id: String): String
+    ProcessInstanceVariablesUpdate(id: String, variables: String): String
```
Comment (cristianonicolai) [inline data-index/data-index-service/data-index-service-common/src/main/resources/schema.graphqls]: I think we need some consistency with the naming, if we keep the main object as prefix, I think we should have something like:
```suggestion
    ProcessInstanceUpdateVariables(id: String, variables: String): String
    NodeInstanceTrigger,
    NodeInstanceRetrigger
    NodeInstanceCancel
```

File: data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java (hunk 1 above)
Comment (nmirasch) [reply data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java]: removed the try/catch to keep consistency between different kinds of calls (post/get/delete..)

File: data-index/data-index-service/data-index-service-common/src/main/resources/schema.graphqls (hunk 2 above)
Comment (nmirasch) [reply data-index/data-index-service/data-index-service-common/src/main/resources/schema.graphqls]: changed!

**Comment by nmirasch** [comment]
@cristianonicolai @radtriste I added the changes suggested. Could you take another look here? Thanks!

**Review by cristianonicolai — APPROVED** [review]

**Review by radtriste — APPROVED** [review]

**Pull request merged**